website/content/docs/glossary.mdx

```diff
@@ -49,6 +50,31 @@ returns a client token which is used for future requests. The token is used by
 Vault to verify the identity of the client and to enforce the applicable ACL
 policies. This token is passed via HTTP headers.
 
+### Plugin
+
+Plugins are a feature of Vault that can be enabled, disabled, and customized to
+some degree. All Vault [auth methods](/docs/auth) and [secrets engines](/docs/secrets)
+are considered plugins.
+
+#### Built-in Plugin
+
+Built-in plugins are shipped with Vault, often for commonly used
+implementations, and require no additional operator intervention to run.
+Built-in plugins are just like any other backend code inside Vault.
+
+#### External Plugin
+
+External plugins are not shipped with Vault and require additional operator
+intervention to run. Vault's external plugins are completely separate,
+standalone applications that Vault executes and communicates with over RPC.
+Each time a Vault secret engine or auth method is mounted, a new process is
+spawned.
+
+#### Multiplexed Plugin
```

Comment:
Should we just define "Multiplexing" instead of "Multiplexed plugin"? I could see potential for someone reading this list and takes it to mean a 3rd separate category (built-in, external, multiplexed) rather than a feature.

Like so:
#### Multiplexing
A feature that allows some plugins to use a single plugin process for multiple mounts of the same time.

Thoughts?